Dashboard/app/js/lib/views/data/inspect-data-table-views.js

```diff
@@ -215,9 +215,33 @@ FLOW.inspectDataTableView = FLOW.View.extend({
 FLOW.DataItemView = FLOW.View.extend({
   tagName: 'span',
   deleteSI: function () {
-    var SI;
+    var SI,slKey, sl;
     SI = FLOW.store.find(FLOW.SurveyInstance, this.content.get('keyId'));
     if (SI !== null) {
+    	// check if we also have the data point loaded locally
+    	slKey = SI.get('surveyedLocaleId');
+    	SL = FLOW.store.filter(FLOW.SurveyedLocale,function(item){
```

Comment:
This is just to make sure that the UI is consistent - which means that I am only interested in surveyedLocales that are actually present in the dashboard, as those might be shown. The code determines if the Locale should be removed from the present UI. If the locale is not present in the dashboard data store, I don't need to worry about it, as it will get updated and reloaded by the delete actions.